internal/component/pyroscope/write/write.go

```diff
@@ -318,6 +330,60 @@ func (f *fanOutClient) Append(ctx context.Context, lbs labels.Labels, samples []
 	return err
 }
 
+type PyroscopeWriteError struct {
+	StatusCode int
+}
+
+func (e *PyroscopeWriteError) Error() string {
+	return fmt.Sprintf("pyroscope write error: status %d", e.StatusCode)
+}
+
+// AppendIngest implements the pyroscope.Appender interface.
+func (f *fanOutClient) AppendIngest(ctx context.Context, profile *pyroscope.IncomingProfile) error {
+	var buf bytes.Buffer
+	tee := io.TeeReader(profile.Body, &buf)
+
+	for i, endpoint := range f.config.Endpoints {
+		u, err := url.Parse(endpoint.URL)
+		if err != nil {
+			return fmt.Errorf("parse endpoint URL: %w", err)
+		}
+
+		u.Path = path.Join(u.Path, profile.URL.Path)
+		u.RawQuery = profile.URL.RawQuery
+
+		var bodyReader io.Reader
+		if i == 0 {
+			bodyReader = tee
+		} else {
+			bodyReader = bytes.NewReader(buf.Bytes())
```

Comment:
Can we avoid reading the whole request body in memory and instead stream it? Maybe in similar way it is handled in the new receiver component with pipes.

This may be especially important after pyroscope v2 is completed since we block requests for 0.5 seconds there.